lib/Twig/Profiler/Profile.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) 2015 Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class Twig_Profiler_Profile implements \IteratorAggregate
```

Comment:
Symfony is serializing this object to store it in the Symfony profiler. So I suggest implementing Serializable here, to control the way it gets serialized (and ensuring BC on the serialization format to allow deserializing older profiles with new versions of the code). The rule in Symfony is that only Serializable classes are guaranteed to be serializable in a BC way (the only way to ensure it for non Serializable classes would be to freeze the class, forbidding to change any private property, which is a no-go)